src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderFlowValidator.java

```diff
@@ -129,24 +128,30 @@ public Future<Void> checkFundLocationRestrictions(List<CompositePoLine> poLines,
   }
 
   private Future<Void> validateLocationRestrictions(CompositePoLine poLine, List<Fund> funds) {
-    List<String> polLocationIds = poLine.getLocations().stream().map(Location::getLocationId).toList();
-    for (Fund fund : funds) {
-      if (Boolean.TRUE.equals(fund.getRestrictByLocations()) && !CollectionUtils.containsAll(fund.getLocationIds(), polLocationIds)) {
-        String poLineId = poLine.getId();
-        String fundId = fund.getId();
-        Collection<String> restrictedLocations = CollectionUtils.subtract(polLocationIds, fund.getLocationIds());
-        logger.error("For POL {} fund {} is restricted to be used for locations {}", poLineId, fundId, restrictedLocations);
-        List<Parameter> parameters = List.of(
-          new Parameter().withKey("poLineId").withValue(poLineId),
-          new Parameter().withKey("poLineNumber").withValue(poLine.getPoLineNumber()),
-          new Parameter().withKey("fundId").withValue(fundId),
-          new Parameter().withKey("fundCode").withValue(fund.getCode()),
-          new Parameter().withKey("restrictedLocations").withValue(restrictedLocations.toString())
-        );
-        return Future.failedFuture(new HttpException(422, ErrorCodes.FUND_LOCATION_RESTRICTION_VIOLATION, parameters));
-      }
+    List<String> restrictedLocations = poLine.getLocations()
+      .stream()
+      .map(Location::getLocationId)
+      .filter(locId -> isRestrictedByAllFunds(funds, locId))
+      .toList();
+
+    if (restrictedLocations.isEmpty()) {
+      return Future.succeededFuture();
     }
-    return Future.succeededFuture();
+
+    String poLineId = poLine.getId();
+    logger.error("For POL {} locations {} are restricted to be used by all funds", poLineId, restrictedLocations);
+    List<Parameter> parameters = List.of(
+      new Parameter().withKey("poLineId").withValue(poLineId),
+      new Parameter().withKey("poLineNumber").withValue(poLine.getPoLineNumber()),
```

Comment:
fundId and fundCode now are not used by UI?
Seems it's used (see comments of this story) or contact with Alisher to confirm

Reply:
thanks for bringing this out. We should ask PO to change the error message or propose some workaround

Reply:
I got it, all funds should match, so we can not provide particular fund code